arangod/RocksDBEngine/RocksDBCollection.cpp

```diff
@@ -679,7 +688,6 @@ std::shared_ptr<Index> RocksDBCollection::createIndex(VPackSlice info,
     RECURSIVE_WRITE_LOCKER(_indexesLock, _indexesLockWriteOwner);
     removeIndex(_indexes, newIdx->id());
```

Comment:
What about this part of cleanup?

Reply:
This is needed only if emplace into `_indexes` happened - e.g. after  Step 3 and that is already handled correct. I think moving it to `indexCleanup` would just introduce complexity but give no benefit.